### PR TITLE
docs: changes to delete oauth_2_client_scopes api

### PR DESCRIPTION
**Pull request by AdithyaSSup**
Added flow diagram and made spec changes to `delete oauth2clientscopes` api contract

### CODE REVIEW

File: api_spec.yaml [hunk 1]
```diff
@@ -576,7 +576,8 @@ paths:
         - OAuth2 Recipe
       operationId: addOAuth2ScopeToClient
```
Comment (rishabhpoddar) [inline api_spec.yaml:577]: Semantic error at paths./appid-<appId>/recipe/oauth2/client/scope.delete.requestBody

File: api_spec.yaml [hunk 2]
```diff
@@ -576,7 +576,8 @@ paths:
         - OAuth2 Recipe
```
Comment (rishabhpoddar) [inline api_spec.yaml:576]: flow diagrams:
- remove `if scopes contains 'supertokens.' prefix || scope ==  'openid'` in POST API.
- reserved scopes can be deleted from clients
- flow for remove scopes:
   - start transactions
   - select for update the client
   - delete

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: api_spec.yaml (hunk 2 above)
Comment (AdithyaSSup) [reply api_spec.yaml:576]: changes made

File: api_spec.yaml (hunk 1 above)
Comment (AdithyaSSup) [reply api_spec.yaml:577]: not able to reproduce in swagger.

Comment (rishabhpoddar) [reply api_spec.yaml:577]: this is till not solved

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: api_spec.yaml (hunk 1 above)
Comment (AdithyaSSup) [reply api_spec.yaml]: issue resolved.

**Pull request merged**